task_list_app/task_list_app/src/main/java/com/jackson/task_list_app/api/controller/TaskController.java

```diff
@@ -24,13 +26,26 @@ public class TaskController {
 
     @GetMapping("/tasks")
     @ResponseStatus(HttpStatus.OK)
-    public List<MyTask> getTask() {
+    public List<Task> getTask() {
         return taskService.getTask();
     }
 
     @PostMapping("/tasks")
-    public ResponseEntity<MyTask> createMyTask(@RequestBody MyTask newMyTask) {
-        MyTask createdMyTask = taskService.createMyTask(newMyTask);
-        return new ResponseEntity<>(createdMyTask, HttpStatus.CREATED);
+    public ResponseEntity<Task> createTask(@RequestBody Task newTask) {
+        Task createdTask = taskService.createTask(newTask);
+        return new ResponseEntity<>(createdTask, HttpStatus.CREATED);
+    }
+
+    @PutMapping("/tasks/{id}") // Adjust the path variable according to your requirements
+    public ResponseEntity<Task> updateTask(@PathVariable Long id, @RequestBody Task updatedTask) {
+        Task existingTask = taskService.getTaskById(id);
+
+        if (existingTask != null) {
+            updatedTask.setId(id);
+            Task task = taskService.updateTask(updatedTask);
+            return new ResponseEntity<>(task, HttpStatus.OK);
```

Comment:
I think we talked about this returning an HTTP 201 Created on success. If this pull request is supposed to finish out card 201, can you please change the PUT return status to 201?

Reply:
done